pkg/minishift/update/update.go

```diff
@@ -264,12 +267,34 @@ func extractBinary(downloadedArchivePath, archiveDir string) (string, error) {
 			return "", err
 		}
 
-		binaryPath = filepath.Join(extract, "minishift")
+		binaryPath, err = minishiftBinaryPath(extract, "minishift")
+		if err != nil {
+			return "", err
+		}
 	}
 
 	return binaryPath, nil
 }
 
+// minishitBinaryPath take extracted Path and os specific binary name and return
+// absolute path of the binary
+func minishiftBinaryPath(extractPath, binaryName string) (string, error) {
+	var binaryPath string
+	var err error
+	fileList := []string{}
+	err = filepath.Walk(extractPath, func(path string, f os.FileInfo, err error) error {
+		fileList = append(fileList, path)
+		return nil
+	})
+	for _, fileName := range fileList {
+		if strings.Contains(fileName, binaryName) {
```

Comment:
I doubt we can use this login in `filepath.Walk...` invocation above and avoid this loop. 

What are the content of `fileList` ?